test/roots.jl

```diff
@@ -62,7 +62,7 @@ end
 
     # Bisection
     rts = roots(f, X, Bisection, 1e-3)
-    @test length(rts) == 4
```

Comment:
I do get 4 boxes here (even though there are only 2 roots).

Reply:
Yes, it gives an incorrect answer. Marking it as `@test_broken` makes transparent that `Bisection` is unreliable and probably should never be used in practise. In addition, it will notify us if `Bisection` ever starts to give the correct result here. (I admit, this is not really related to the topic of this PR–I think I just noticed this inconsistency while I was working on it.)

Reply:
I see what you mean, but I don't agree that the result is unreliable. It is reliably giving you a set of intervals that are guaranteed to contain all of the roots in that box, and correctly telling you that it is unable to guarantee whether or not there are actually roots there.

The test is testing what the function returns, not what we would like it to ideally return ;)

Reply:
Ok, that makes sense. But testing for the result 4 seems overly specific if any number above 2 would acceptable. Maybe it would be better to test that the exact solutions are contained in whatever boxes are returned? I've created a separate PR #156 for that.